math/complex_numbers.cpp

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2020 @author tjgurwara99
+ * @file
+ *
+ * A basic implementation of Complex Number field as a class with operators
+ * overloaded to accommodate (mathematical) field operations.
+ */
+
+#include <cmath>
+#include <iostream>
+#include <stdexcept>
+#include <cassert>
+
+/**
+ * Class Complex to represent complex numbers as a field.
+ */
+class Complex {
+    // The real value of the complex number
+    double re;
+    // The imaginary value of the complex number
+    double im;
+
+  public:
+    /**
+     * Complex Constructor which initialises the complex number which takes two
+     * arguments.
+     * @param x The real value of the complex number.
+     * @param y The imaginary value of the complex number.
+     */
+    Complex(double x, double y) {
```

Comment:
C++11 allows us to instantiate with default values. Also, `this` keyword incorrect as the instance is not complete yet. The correct way is:
```suggestion
    Complex(double x 0 0.f, double y = 0.f) : re(x), im(y) {;
```